webapp/store/posts.js

```diff
@@ -120,26 +114,4 @@ export const getters = {
   orderIcon(state) {
     return state.order.icon
   },
-  currentPosts(state) {
-    return state.currentPosts || []
-  },
-}
-
-export const actions = {
-  async refreshPosts({ commit, getters }, { i18n }) {
-    const client = this.app.apolloProvider.defaultClient
-    const {
-      data: { Post },
-    } = await client.query({
-      query: filterPosts(i18n),
-      variables: {
-        filter: getters.filter,
-        first,
-        orderBy: ['pinned_asc', getters.orderBy],
-        offset,
-      },
-    })
-    commit('SET_CURRENT_POSTS', Post)
-    return Post
-  },
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-15T13:39:29Z" title="Friday, November 15th 2019, 2:39:29 pm +01:00">Nov 15, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -120,26 +114,4 @@ export const getters = {
   orderIcon(state) {
     return state.order.icon
   },
-  currentPosts(state) {
-    return state.currentPosts || []
-  },
-}
-
-export const actions = {
-  async refreshPosts({ commit, getters }, { i18n }) {
-    const client = this.app.apolloProvider.defaultClient
-    const {
-      data: { Post },
-    } = await client.query({
-      query: filterPosts(i18n),
-      variables: {
-        filter: getters.filter,
-        first,
-        orderBy: ['pinned_asc', getters.orderBy],
-        offset,
-      },
-    })
-    commit('SET_CURRENT_POSTS', Post)
-    return Post
-  },
```


Get rid of the store :+1: 

Did you know that `vue-apollo` suggests to use it's cache as a substitute for `vuex`?
https://vue-apollo.netlify.com/guide/local-state.html#why-use-apollo-local-state-management

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-15T14:16:40Z" title="Friday, November 15th 2019, 3:16:40 pm +01:00">Nov 15, 2019</time>_

---


yeah, but how to do this when you are trying to reset the cache and refetch Posts from a `layout` to a `page`? I'm quite sure they don't have examples for this

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-15T14:39:13Z" title="Friday, November 15th 2019, 3:39:13 pm +01:00">Nov 15, 2019</time>_

---


:thinking: local state is for state that is not stored in the backend. So `refetch` is certainly out of scope of the local state.

@mattwr18 why does the layout has to set posts for the page component :astonished: ?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-15T16:03:26Z" title="Friday, November 15th 2019, 5:03:26 pm +01:00">Nov 15, 2019</time>_

---


because they want to click on the logo which lives in the `default.vue` and it refetch the Posts...